src/sas/sascalc/dataloader/data_info.py

```diff
@@ -114,8 +114,12 @@ def __init__(self, data=None, err_data=None, qx_data=None,
         self.qx_data = np.asarray(qx_data)
         self.qy_data = np.asarray(qy_data)
         self.q_data = np.asarray(q_data)
-        self.mask = np.asarray(mask)
-        self.err_data = np.asarray(err_data)
+        if mask is not None:
+            self.mask = np.asarray(mask)
+        else:
+            self.mask = np.ones(self.data.shape, dtype=bool)
+        if err_data is not None:
+            self.err_data = np.asarray(err_data)
```

Comment:
self.mask and self.err_data get defined only when mask/err_data is passed as arguments.
Subsequent calls to _validity_check (line 1055) will fail with attribute errors if those optional arguments are not passed.

Reply:
Good catch on this. self.err_data needs to be set to something for uncertainty propagation when combining data sets. I want to say self.err_data should be set to all zeros, but something about that doesn't seem quite right.

self.mask is set to an array of trues when no mask is passed, so that should be good already.